pkg/crc/cache/cache.go

```diff
@@ -18,22 +20,24 @@ type Cache struct {
 	binaryName string
 	archiveURL string
 	destDir    string
+	version    string
+	versionArg []string
 }
 
-func New(binarName string, archiveURL string, destDir string) *Cache {
-	return &Cache{binaryName: binarName, archiveURL: archiveURL, destDir: destDir}
+func New(binarName string, archiveURL string, destDir string, version string, versionArg []string) *Cache {
+	return &Cache{binaryName: binarName, archiveURL: archiveURL, destDir: destDir, version: version, versionArg: versionArg}
 }
 
-func NewOcCache(destDir string) *Cache {
-	return New(constants.OcBinaryName, constants.GetOcUrl(), destDir)
+func NewOcCache(destDir string, version string, versionArg []string) *Cache {
```

Comment:
I should have thought of this before, but I don't think having `destDir` as a `NewOcCache` parameter is really useful, we want the `oc` binary to go to a welll known location, we won't have some parts of the code putting it in /some/dir, and other parts of the code putting it in /other/dir. We could add a `GetBinaryPath()` or such to this package.

Reply:
@cfergeau this makes it really explicit, we we add `GetBinaryPath()` then how you want to handle podman, oc  or any other future binary which will have it's own path? or you want to provide the path as per binaryName?

Reply:
`func (c *Cache) GetBinaryPath() (string, error)` this would be a method, so it would be `ocCache.GetBinaryPath()`, `podmanCache.GetBinaryPath()`, ... It would not do a lot more than `return filepath.join(c.DestDir, c.BinaryName)` (approximate syntax/implementation :)

Reply:
@cfergeau `return filepath.join(c.DestDir, c.BinaryName)` even here we do need `destDir` and how do we get it?

Reply:
```
diff --git a/pkg/crc/cache/cache.go b/pkg/crc/cache/cache.go
index 72a5babc..6b54f826 100644
--- a/pkg/crc/cache/cache.go
+++ b/pkg/crc/cache/cache.go
@@ -24,8 +24,8 @@ func New(binarName string, archiveURL string, destDir string) *Cache {
        return &Cache{binaryName: binarName, archiveURL: archiveURL, destDir: destDir}
 }
 
-func NewOcCache(destDir string) *Cache {
-       return New(constants.OcBinaryName, constants.GetOcUrl(), destDir)
+func NewOcCache() *Cache {
+       return New(constants.OcBinaryName, constants.GetOcUrl(), constants.CrcOcBinDir)
 }
 
 func NewPodmanCache(destDir string) *Cache {
@@ -36,6 +36,10 @@ func NewGoodhostsCache(destDir string) *Cache {
        return New(constants.GoodhostsBinaryName, constants.GetGoodhostsUrl(), destDir)
 }
 
+func (c *Cache) GetBinaryPath() string {
+       return filepath.Join(c.destDir, c.binaryName)
+}
+
 func (c *Cache) IsCached() bool {
        if _, err := os.Stat(filepath.Join(c.destDir, c.binaryName)); os.IsNotExist(err) {
                return false
diff --git a/pkg/crc/preflight/preflight_checks_common.go b/pkg/crc/preflight/preflight_checks_common.go
index 0fa00942..5f22d5d1 100644
--- a/pkg/crc/preflight/preflight_checks_common.go
+++ b/pkg/crc/preflight/preflight_checks_common.go
@@ -71,7 +71,7 @@ func checkOcBinaryCached() error {
        // We should remove this code after 3-4 releases. (after 2020-07-10)
        os.Remove(filepath.Join(constants.CrcBinDir, "oc"))
 
-       oc := cache.NewOcCache(constants.CrcOcBinDir)
+       oc := cache.NewOcCache()
        if !oc.IsCached() {
                return errors.New("oc binary is not cached")
        }
@@ -80,7 +80,7 @@ func checkOcBinaryCached() error {
 }
 
 func fixOcBinaryCached() error {
-       oc := cache.NewOcCache(constants.CrcOcBinDir)
+       oc := cache.NewOcCache()
        if err := oc.EnsureIsCached(); err != nil {
                return fmt.Errorf("Unable to download oc %v", err)
        }
```
really 2 separate suggestions, we should imo be removing `destDir` from `NewOcCache` and similar methods. The `GetBinaryPath()` suggestion is more a "would such an API be useful?" question.